### PR TITLE
WIP: Code tweaks

### PR DESCRIPTION
**Pull request by ultrasecreth**
- Use a struct fro the meta params in the aave wrapper
- Use a struct fro the meta params in the balancer wrapper
- Use a struct fro the meta params in the uniswap  wrapper

### CODE REVIEW

File: src/BaseWrapper.sol [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        // Avoid storage write if not needed
+        if (result.length > 0) {
+            delete _callbackResult;
+        }
+        return result;
+    }
+
+    function _flashLoan(IERC20 asset, uint256 amount, bytes memory params) internal virtual;
+
+    function _handleFlashLoan(IERC20 asset, uint256 amount, uint256 fee, bytes memory params) internal {
```
Comment (alcueca) [inline src/BaseWrapper.sol:52]: Maybe?

```suggestion
    function _bridgeToCallback(IERC20 asset, uint256 amount, uint256 fee, bytes memory params) internal {
```

File: src/uniswapV3/UniswapV3Wrapper.sol [hunk 2]
```diff
@@ -57,97 +80,44 @@ contract UniswapV3Wrapper is IERC3156PPFlashLender, IUniswapV3FlashCallback {
      * @return The amount of `asset` to be charged for the loan, on top of the returned principal.
      */
     function flashFee(IERC20 asset, uint256 amount) public view override returns (uint256) {
-        address pool = address(getPool(asset));
```
Comment (alcueca) [inline src/uniswapV3/UniswapV3Wrapper.sol:60]: The natspec above can be copied from a wrapper that has it correct.

File: src/BaseWrapper.sol [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (alcueca) [inline src/BaseWrapper.sol:1]: We need to use licenses consistently, and copy the right one to LICENSE.md.

GPL or MIT?

File: src/BaseWrapper.sol [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    struct Data {
+        address loanReceiver;
+        address initiator;
+        function(address, address, IERC20, uint256, uint256, bytes memory) external returns (bytes memory) callback;
+        bytes initiatorData;
+    }
+
+    bytes internal _callbackResult;
+
+    /// @inheritdoc IERC3156PPFlashLender
```
Comment (alcueca) [inline src/BaseWrapper.sol:22]: We should @inheritdoc from the ERC7399 interface when possible, I think.

**Review by alcueca — APPROVED** [review]
Only minor issues related to natspec and function naming.

File: src/BaseWrapper.sol (hunk 3 above)
Comment (ultrasecreth) [reply src/BaseWrapper.sol:1]: MIT I guess?

File: src/BaseWrapper.sol (hunk 4 above)
Comment (ultrasecreth) [reply src/BaseWrapper.sol:22]: Where does that interface lives? I can't see it in the deps

**Pull request merged**